src/main.rs

```diff
@@ -196,6 +197,7 @@ Then try to restart Plume
         }
     });
 
+    let mut valid_domains: HashMap<&str, Instant> = HashMap::new();
```

Comment:
I'd be in favor of using the newtype design pattern for this, so that it make more sense what it is, even at the type level.
I don't know where are coming the &str from, but you might want to use String instead, or you'll have some difficulties with lifetime

Reply:
aye. i haven't gotten that far yet, because i haven't implemented the other side of this